File: evd_ros_backend/evd_ros_core/src/evd_script/program/primitive.py
from ..node import Node


class Primitive(Node):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls, trailing_delim=True):
        return 'primitive' + '.' if trailing_delim else ''

    @classmethod
    def full_type_string(cls):
        return Node.full_type_string() + cls.type_string()

    def __init__(self, type='', name='', uuid=None, parent=None, append_type=True):
        super(Primitive,self).__init__(
            type=Primitive.type_string() + type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type)

File: evd_authoring_app/src/model/evdScript/program/primitive.js
import { Node } from '../node';


export class Primitive extends Node {

    /*
    * data structure methods
    */

    static typeString() {
        return 'primitive.';
    }

    static fullTypeString() {
        return Node.fullTypeString() + Primitive.typeString();
    }

    constructor(type='', name='', uuid=null, parent=null, appendType=true) {
        super(
            (appendType) ? 'primitive.'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new Primitive(
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false
        );
    }

}